### DISCUSSION TITLE
Clicking predefined route Link causes page reload

### DISCUSSION THREAD

**Discussion by TheoMer** [vercel/next.js#21590] [Help] [answered]
OS: Deployed to Heroku
Next: 10.0.6-canary.7
Browsers: Chrome Version 88.0.4324.104 (Official Build) (64-bit) and Version 88.0.4324.104 (Official Build) beta (64-bit)

So I have two types of Link url:

1. frontend/components/Nav.tsx
```
    <Link href="/items" passHref>
      <a>SHOP</a>
    </Link>
```

2. frontend/components/Item.tsx
```
    <Link 
      href={{
        pathname: '/item',
        query: { id: item.id }
      }} passHref>
      <a>{item.image && <img src={item.image} alt={item.title} />}</a>
    </Link>
```

The site functions properly when any of the menu links are clicked. But as soon as I click one of the item links, on the main page, every link clicked thereafter will cause a reload of the page. And this is with caching specified in frontend/server.tsx  as follows:

```
const ssrCache = cacheableResponse({
  ttl: 1000 * 60 * 60, // 1hour
  get: async ({ req, res }) => {
    const data = await app.render(req, res, req.path, {
      ...req.query,
      ...req.params,
    })

    // Add here custom logic for when you do not want to cache the page, for
    // example when the page returns a 404 status code:
    if (res.statusCode === 404) {
      res.end(data)
      return
    }

    return { data }
  },
  send: ({ data, res }) => res.send(data),
})

app.prepare().then(() => {
  server.get('/', (req, res) => ssrCache({ req, res }))

  server.get('/item?id=:id', (req, res) => {
    return ssrCache({ req, res })
  })

  server.get('/order?id=:id', (req, res) => {
    return ssrCache({ req, res })
  })
}
```

What is the issue here?

Sample code can be found here: https://github.com/TheoMer/Prisma2-Migration/tree/master/frontend
My site: https://frontend.theomer.co.uk/

**Comment by TheoMer** [comment, marked as answer]
The issue has been resolved.

**Reply by stigkj** [reply]
Maybe you could mark this as the answer? And even better if you could say how it was fixed 😀
